docs/tutorials/user/credit-class-project-batch-management.md

```diff
@@ -0,0 +1,468 @@
+
+If you are managing your own credit origination process, you need to host your own data. If you are not ready to figure out a solution for hosting data but you are ready to create a credit class, project, and batch, then feel free to skip to the [next section](#credit-class) and come back later.
+
+To make your data available in Regen Network Development applications, you need to use the same IRI generation method we used in the previous section and create a data resolver using the `data` module that points applications to the hosted data when provided the IRI of the data.
+
+For more information about the `data` module, see [Data Concepts](../../modules/data/01_concepts.md).
+
+### Define Resolver
+
+The following command will create a data resolver with a url of `[url]`. This is the URL at which you are hosting the data. You are claiming that given an IRI (e.g. `[url] + [iri]`), an application can fetch the data in complete or partial form depending on how you manage privacy.
+
+To create a data resolver, run the following command:
+
+```sh
+regen tx data define-resolver [url]
+```
+
+For more information about the command, add `--help` or check out [the docs](../../commands/regen_tx_data_define-resolver.md).
+
+Now that you created a data resolver, you can query it by id:
+
+```sh
+regen q data resolver [id]
+```
+
+### Content Hash
+
+Before we can anchor and register the data we created in the previous section, we must convert our IRIs into JSON objects representing content hashes on chain.
+
+To convert an IRI to a content hash, run the following command:
+
+```sh
+regen q data convert-iri-to-hash [iri]
+```
+
+For more information about the command, add `--help` or check out [the docs](../../commands/regen_query_data_convert-iri-to-hash.md).
+
+### Register Resolver
+
+The data for each credit class, project, and batch can now be registered to the resolver. The account that created the resolver is the only account that can register data to the resolver.
+
+With the next command, you can register all your data at once. You can register credit class, project, and batch data using the content hashes we converted from IRIs in the last step.
+
+To register data to a resolver, run the following command:
+
+```sh
+regen tx data register-resolver [resolver-id] [content-hashes-json]
+```
+For more information about the command, add `--help` or check out [the docs](../../commands/regen_tx_data_register-resolver.md).
+
+Now that you registered data to the resolver, you can look up resolvers by IRI:
+
+```sh
+regen q data resolvers-by-iri [iri]
+```
+
+If the data is in fact hosted at the location the data resolver is pointing to, the data can be fetched by Regen Network Development applications.
+
+## Credit Class
+
+Now that we have data for our class, project, and batch, we have IRIs for each, and we may or may not be hosting the data, we are ready to create a credit class.
```

Comment:
I think this will be easier to read if split up into multiple sentences

Reply:
Can you provide an example? I think this reads fine as is.

Reply:
I didn't make any changes here.